core/controllers/classifier.py

```diff
@@ -0,0 +1,118 @@
+        received_signature: str. The signature received from the VM.
+
+    Returns:
+        bool.
+
+    Raises:
+        UnauthorizedUserException.
+    """
+    secret = None
+    for val in config_domain.VMID_SHARED_SECRET_KEY_MAPPING.value:
+        if val['vm_id'] == vm_id:
+            secret = val['shared_secret_key'][0]
+            break
+    if secret is None:
+        return False
+    secret = str([val['shared_secret_key'] for val in (
+        config_domain.VMID_SHARED_SECRET_KEY_MAPPING.value) if val[
+            'vm_id'] == vm_id][0])
+    generated_signature = hmac.new(
+        secret, message, digestmod=hashlib.sha256).hexdigest()
+    if generated_signature != received_signature:
+        return False
+    if vm_id == feconf.DEFAULT_VM_ID and not feconf.DEV_MODE:
+        return False
+    return True
+
+
+class TrainedClassifierHandler(base.BaseHandler):
+    """This handler stores the result of the training job in datastore and
+    updates the status of the job.
+    """
+
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    def post(self):
+        """Handles POST requests."""
+        payload = self.request.get('payload')
+        payload = json.loads(payload)
+        signature = payload.get('signature')
+        message = payload.get('message')
+        vm_id = payload.get('vm_id')
+        message = json.dumps(message, sort_keys=True)
+        if verify_signature(message, vm_id, signature):
+            job_id = self.payload.get('message').get('job_id')
+            classifier_data = self.payload.get('message').get('classifier_data')
+            classifier_training_job = (
+                classifier_services.get_classifier_training_job_by_id(job_id))
+            state_name = classifier_training_job.state_name
+            exp_id = classifier_training_job.exp_id
+            exp_version = classifier_training_job.exp_version
+            algorithm_id = classifier_training_job.algorithm_id
+
+            data_schema_version = None
+            for algorithm_details in (
+                    feconf.INTERACTION_CLASSIFIER_MAPPING.values()):
+                if algorithm_details['algorithm_id'] == algorithm_id:
+                    data_schema_version = algorithm_details[
+                        'current_data_schema_version']
+            if data_schema_version is None:
+                raise self.InternalErrorException
+
+            classifier = classifier_domain.ClassifierData(
+                job_id, exp_id, exp_version, state_name, algorithm_id,
+                classifier_data, data_schema_version)
+            try:
+                classifier_services.save_classifier(classifier)
+            except Exception:
+                raise self.InternalErrorException
+
+            # Update status of the training job to 'COMPLETE'.
+            classifier_training_job.update_status(
+                feconf.TRAINING_JOB_STATUS_COMPLETE)
+            classifier_services.save_classifier_training_job(
```

Comment:
@seanlip This just updates the job. I don't see why thats an issue. Nothing in the job class is mutable except status. So, if anytime the update function is triggered, it will be for this purpose.